### PR TITLE
Phoenix6 IsConnected -> master 

### PR DESCRIPTION
**Pull request by drorta**

### CODE REVIEW

**Review by YHerm — CHANGES_REQUESTED** [review]
a lot of duplicate code, find a better way

File: src/main/java/frc/robot/hardware/phoenix6/Phoenix6Device.java [hunk 1]
```diff
@@ -27,12 +28,14 @@ public Phoenix6Device(String logPath) {
 		AlertManager.addAlert(new PeriodicAlert(Alert.AlertType.ERROR, logPath + "disconnectedAt", () -> !isConnected()));
 	}
 
+	public abstract ParentDevice getDevice();
```
Comment (YHerm) [inline src/main/java/frc/robot/hardware/phoenix6/Phoenix6Device.java:31]: abstract funcs in bottom of class

File: src/main/java/frc/robot/hardware/phoenix6/gyro/Pigeon2Gyro.java [hunk 2]
```diff
@@ -19,4 +20,9 @@ public void setYaw(Rotation2d yaw) {
 		gyro.setYaw(yaw);
 	}
 
+	@Override
+	public ParentDevice getDevice() {
```
Comment (YHerm) [inline src/main/java/frc/robot/hardware/phoenix6/gyro/Pigeon2Gyro.java:24]: make it return Pigeon2

File: src/main/java/frc/robot/hardware/phoenix6/angleencoder/CANCoderEncoder.java [hunk 3]
```diff
@@ -20,4 +21,9 @@ public void setPosition(Rotation2d position) {
 		encoder.setPosition(position.getRotations());
 	}
 
+	@Override
+	public ParentDevice getDevice() {
```
Comment (YHerm) [inline src/main/java/frc/robot/hardware/phoenix6/angleencoder/CANCoderEncoder.java:25]: make it return cancoder

File: src/main/java/frc/robot/hardware/phoenix6/motor/TalonFXMotor.java [hunk 4]
```diff
@@ -101,4 +102,9 @@ public void applyRequest(IRequest<?> request) {
 		}
 	}
 
+	@Override
+	public ParentDevice getDevice() {
+		return motor;
+	}
```
Comment (YHerm) [inline src/main/java/frc/robot/hardware/phoenix6/motor/TalonFXMotor.java:108]: make it return talonfx

**Review by YHerm — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/hardware/phoenix6/angleencoder/CANCoderEncoder.java
Comment (YHerm) [inline src/main/java/frc/robot/hardware/phoenix6/angleencoder/CANCoderEncoder.java:1]: i'm pretty sure this function already exist in this class (also in the talonfxmotor and pigeon2gyro)

**Review by YHerm — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/hardware/phoenix6/gyro/Pigeon2Gyro.java [hunk 5]
```diff
@@ -19,4 +20,9 @@ public void setYaw(Rotation2d yaw) {
 		gyro.setYaw(yaw);
 	}
 
+	@Override
+	public Pigeon2 getDevice() {
```
Comment (hillelv) [inline src/main/java/frc/robot/hardware/phoenix6/gyro/Pigeon2Gyro.java:24]: why not the wrapper? either return wrapper or parent device, no need to choose a class in between

**Review by YHerm — APPROVED** [review]

**Review by Nitay4 — APPROVED** [review]
why so skibidy

**Review by hillelv — APPROVED** [review]

**Review by tnrz135 — APPROVED** [review]

**Pull request merged**